Formula/smug.rb

```diff
@@ -0,0 +1,17 @@
+class Smug < Formula
+  desc "Automate your tmux workflow"
+  homepage "https://github.com/ivaaaan/smug"
+  url "https://github.com/ivaaaan/smug/archive/refs/tags/v0.2.5.tar.gz"
+  sha256 "314711f616ea71613fe883fe579e7d3bc0051f540f27c7925a1ce6a0ece69378"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "-ldflags", "-X main.version=#{version}"
+  end
+
+  test do
+    assert_equal(version, shell_output("smug").lines.first.split("Version").last.chomp)
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
I did my best here - the difficulty is smug is meant to be used interactively, to spawn tmux instances. That seems really error prone, and I think it might even be dangerous (for example, if you have a tmux.conf that launches other things, that is now available to the `test` block).

I appreciate guidance and examples of doing this safely and better.

Reply:
The test block shouldn't know about any local configuration so I don't expect this to happen. (we set `$HOME` and `$USER` to something new here)